pup-0003.md

```diff
@@ -0,0 +1,157 @@
+feature, bug fix, or hotfix.
+
+#### New Feature
+
+1. All feature PRs are opened and merged against master.
+
+#### Bug Fix
+
+1. All bug fix PRs are opened against master
+2. After the PR is ready to be merged, the person merging the commit would
+   first merge the commit to master. Then, this person cherry-picks this change
+   from master to the latest x.y-dev branch.
+3. The person cherry-picking would also need to track in Redmine which dev
+   branch(es) the change has been applied to.
+
+#### Hotfix
+
+The current hotfix process[2] would remain basically unchanged. However,
+instead of merging forward the commit, we'd use cherry-picking.
+
+1. The affected release would be branched and a hotfix would be applied.
+2. After the release, the hotfix gets cherry-picked into the x.y-release branch.
+3. The change is then cherry-picked into the x.y-dev branch and master.
+4. The person cherry-picking would also need to track in Redmine which release
+   branches the change has been made to.
+
+### Conflicts
+
+If the cherry-pick has conflicts, we should consider abandoning the cherry-pick
+and encourage users to upgrade to the next Y release.
+
+### Using git cherry-pick
+
+When we perform the cherry-pick in git, we should use the `-x` flag which adds
+the original commit id to the commit message of the cherry-pick. This will help
+track changes across branches.
+
+### Updating Redmine
+
+When a cherry-pick is pushed to a dev branch, the release field of the issue in
+Redmine should be set to x.y.next. For example, when a cherry-pick is pushed to
+2.13-dev, the issue's release field should be set to 2.13.next. Then when the
+change is released, it's field will be updated with the actual value (e.g.
+2.13.4). For a hotfix that is applied to a release branch, the release branch is
+set to the hotfix release version.
+
+### Timeframe
+
+When this proposal is accepted, we'll work with release engineering to determine
+a date to switch to cherry-picking. After a date has been picked, an
+announcement will be sent out to the pulp-dev list with the planned date. After
+tihs date, we should begin to follow this proposal. Also, we'll enter in Redmine
+issues to update our docs according after this proposal has been accepted.
+
+## Drawbacks
+
+Git is really built for changes to be merged between branches. For example,
+when cherry-picking a change, the commit is assigned a new id but when merging
+the commit, it retains the same id. It makes looking up the commit to see what
+branches it's been applied to easier. If we cherry-pick, we have to track this
+information externally to git. This means also having to remember to update
+Redmine.
+
+Switching to this model also creates some inconsistency. For example, if we
+adopt this for Pulp 3, then we still have to use merge forwarding for Pulp 2.
+Then we are stuck having to support two different strategies.
+
+Also, when creating bug fixes in the cherry-pick model, code will be developed
```

Comment:
I agree but I also feel like if we merge forward from dev branches, then we have more time to find changes/merge conflicts on master since there's more time between master being released than a dev branch. Merging forward from a dev branch seems safer in this respect to me. I guess since we also added the part about NOT cherry-picking the change to the dev branch if there are conflicts that my point is somewhat negated. What do you think?